### PR TITLE
showing errors on wrong numbers in PhoneNumberFormField

### LINKED ISSUE #6027: Phone validation errors are not being shown during onBlur
> - Follow up of #5999 
 
> -> Invalid input's post-submission getting rejected without error message
> -> Even though in user filters, we say it says  whatsapp phone number doesn't support landline number, the field is accepting the data and showing the badge
> 
> _Originally posted by @nihal467 in https://github.com/coronasafe/care_fe/issues/5999#issuecomment-1669401821_

**Describe the bug**
When a phone number is entered in advanced filters (eg: user filters), it is not showing invalid phone number if a landline number is entered when the field is supposed to accept only mobile numbers.

**To Reproduce**

1. Go to User Advanced filters
2. Enter landline number in WhatsApp number
3. Change element focus to trigger validation check
4. Does not show validation error message

**Expected behavior**
Should show validation error message

### PR DESCRIPTION
**Pull request by kshitijv256**
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 02ff7fb</samp>

Removed unused `errorClassName` props from some components and simplified error handling logic in user filter and shifting list filter forms. These changes improve the code readability and maintainability.

## Proposed Changes

- Fixes #6027 
- Showing errors when entered incorrect number in PhoneNumberFormField

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

### Screenshot

Screenshot from 2023-11-20 23-31-52

### CODE REVIEW

**Review by rithviknishad — CHANGES_REQUESTED** [review]
The whitespace between the input fields are not consistent now

**Comment by kshitijv256** [comment]
@rithviknishad @nihal467 
Fixed the incorrect spacing problem

Screenshot from 2023-11-25 22-41-34

File: src/Components/Form/FormFields/FormField.tsx [hunk 1]
```diff
@@ -71,7 +71,12 @@ const FormField = ({
         )}
       </div>
       <div className={field?.className}>{children}</div>
-      <FieldErrorText error={field?.error} className={field?.errorClassName} />
+      {field?.error ? (
+        <FieldErrorText
+          error={field?.error}
+          className={field?.errorClassName}
+        />
+      ) : null}
```
Comment (rithviknishad) [inline src/Components/Form/FormFields/FormField.tsx]: Revert this, as this now breaks the consistent padding in all other places

image


```suggestion
      <FieldErrorText error={field?.error} className={field?.errorClassName} />
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]
The right way is to adjust the parent div's `gap-[x]` tailwindcss class of the filters

File: src/Components/Users/UserFilter.tsx [hunk 2]
```diff
@@ -134,26 +134,26 @@ export default function UserFilter(props: any) {
...
-      <PhoneNumberFormField
-        label="Whatsapp Number"
-        name="alt_phone_number"
-        placeholder="WhatsApp Phone Number"
-        value={filterState.alt_phone_number}
-        onChange={handleChange}
-        errorClassName="hidden"
-        types={["mobile"]}
-      />
+      <div className="mb-[-16px]">
```
Comment (rithviknishad) [inline src/Components/Users/UserFilter.tsx]:
```suggestion
      <div className="-mb-4">
```

File: src/Components/Users/UserFilter.tsx [hunk 3]
```diff
@@ -134,26 +134,26 @@ export default function UserFilter(props: any) {
...
+        <PhoneNumberFormField
+          label="Phone Number"
+          name="phone_number"
+          placeholder="Phone Number"
+          value={filterState.phone_number}
+          onChange={handleChange}
+          types={["mobile", "landline"]}
+        />
+      </div>
+      <div className="mb-[-16px]">
```
Comment (rithviknishad) [inline src/Components/Users/UserFilter.tsx]:
```suggestion
      <div className="-mb-4">
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**